grafana-plugin/src/models/alert_receive_channel/alert_receive_channel.ts

```diff
@@ -333,47 +293,37 @@ export class AlertReceiveChannelStore extends BaseStore {
       method: 'DELETE',
     });
 
-    return this.updateChannelFilters(channelFilter.alert_receive_channel, true);
+    return this.fetchChannelFilters(channelFilter.alert_receive_channel, true);
   }
 
-  @action.bound
-  async updateAlertReceiveChannelOptions() {
-    const response = await makeRequest(`/alert_receive_channels/integration_options/`, {});
+  async fetchAlertReceiveChannelOptions() {
+    const { data } = await onCallApi().GET(`/alert_receive_channels/integration_options/`, undefined);
 
     runInAction(() => {
-      this.alertReceiveChannelOptions = response;
+      this.alertReceiveChannelOptions = data;
     });
   }
 
-  getIntegration(alertReceiveChannel: Partial<AlertReceiveChannel>): SelectOption {
-    return (
-      this.alertReceiveChannelOptions &&
-      alertReceiveChannel &&
-      this.alertReceiveChannelOptions.find(
-        (alertReceiveChannelOption: SelectOption) => alertReceiveChannelOption.value === alertReceiveChannel.integration
-      )
-    );
-  }
-
-  @action.bound
   @WithGlobalNotification({ success: 'Integration has been saved', failure: 'Failed to save integration' })
-  async saveAlertReceiveChannel(id: AlertReceiveChannel['id'], data: Partial<AlertReceiveChannel>) {
-    const item = await this.update(id, data, undefined, true);
+  async saveAlertReceiveChannel(
+    id: ApiSchemas['AlertReceiveChannel']['id'],
+    payload: Partial<ApiSchemas['AlertReceiveChannelUpdate']>
+  ) {
+    const { data } = await onCallApi().PUT('/alert_receive_channels/{id}/', {
+      params: { path: { id } },
+      // TODO: fix 400 when inbound_email is missing and get rid of casting
+      body: { heartbeat: payload.heartbeat || null, ...payload } as ApiSchemas['AlertReceiveChannelUpdate'],
```

Comment:
@vadimkerr another thing to check. Frontend provides here only a partial object (only updated properties) so it's more like a PATCH. According to openapi schema backend expects full object (which seems correct because it's PUT). But when I provide here full object, the `inbound_email` is `null` (GET returns it as `null`) and there's a validatin error on PUT saying: "
`inbound_email: ["This field may not be null."]`

If it's `null` on the backend, we shouldn't receive validation error, wdyt?

Reply:
fixed in https://github.com/grafana/oncall/pull/3851/commits/b06f50f4471450faa089fddec4c81f415bb32d94